Source/WebCore/platform/encryptedmedia/CDMProxy.cpp

```diff
@@ -357,7 +357,7 @@ void CDMInstanceProxy::startedWaitingForKey()
     bool wasWaitingForKey = m_numDecryptorsWaitingForKey > 0;
     m_numDecryptorsWaitingForKey++;
 
-    callOnMainThread([player = m_player, wasWaitingForKey] {
+    callOnMainThread([player = m_player.get(), wasWaitingForKey] {
```

Comment:
This is a behavior change, you are now keeping the player alive in the lambda. Why aren’t you capturing the weak pointer like we used to?

Reply:
If we switch it back to a weak reference in the lambda, we just have to grab the strong reference on the main thread inside the lambda:

```
    callOnMainThread([weakPlayer = m_player, wasWaitingForKey] {
        auto player = weakPlayer.get();
        if (player && !wasWaitingForKey)
             player->waitingForKeyChanged();
     });
```

I'll make that change.

Reply:
Fixed this in the second patch.